spec/search_builders/curation_concerns/search_builder_spec.rb

```diff
@@ -0,0 +1,58 @@
+require 'spec_helper'
+
+describe CurationConcerns::SearchBuilder do
+  let(:processor_chain) { [:filter_models] }
+  let(:ability) { double('ability') }
+  let(:context) { double('context') }
+  let(:user) { double('user') }
+  let(:solr_params) { { fq: [] } }
+
+  subject { described_class.new(processor_chain, context) }
+
+  describe '#only_generic_files' do
+    before { subject.only_generic_files(solr_params) }
+
+    it 'adds GenericFile to query' do
+      expect(solr_params[:fq].first).to include('{!raw f=has_model_ssim}GenericFile')
+    end
+  end
+
+  describe '#find_one' do
+    before do
+      allow(subject).to receive(:blacklight_params).and_return(id: '12345')
+      subject.find_one(solr_params)
+    end
+
+    it 'adds id to query' do
+      expect(solr_params[:fq].first).to include('{!raw f=id}12345')
+    end
+  end
+
+  describe '#gated_discovery_filters' do
+    before do
+      allow(subject).to receive(:current_ability).and_return(ability)
+      allow(ability).to receive(:current_user).and_return(user)
+      allow(user).to receive(:groups).and_return(['admin'])
+    end
+
+    it 'does not filter results for admin users' do
+      expect(subject.gated_discovery_filters).to eq([])
+    end
+  end
+
+  describe '#discovery_permissions' do
+    before { allow(subject).to receive(:blacklight_params).and_return(works: 'mine') }
+
+    it 'limits query to edit permissions' do
```

Comment:
I see. It's only when params[:works] = 'mine'.  Can we put this in a `context "when showing my works"` block?